druid/druid-api/src/services/ValidationService.ts

```diff
@@ -0,0 +1,194 @@
+import _ from "lodash";
+import { ICommon, IFilter, ILimits, IQuery, IRules, IValidationResponse } from "../models/models";
+import { APILogger } from "./ApiLogger";
+
+/**
+ * Service to validate/filter/limits the user druid queries.
+ */
+export class ValidationService {
+    /**
+     * Method to validate the user druid query for every queryType.
+     * @param query IQuery - User druid query/request.
+     * @param limits ILimits - Limit/filter Configurations to validate the queries.
+     */
+    public static validate(query: IQuery, limits: ILimits): IValidationResponse {
+        // If the limit is exceeded than than the config then set to default.
+        const commonRulesValidationStatus: IValidationResponse = this.validateCommonRules(query, limits.common);
+        if (commonRulesValidationStatus.isValid) {
+            switch (query.queryType.toLowerCase()) {
+                case "groupby": return this.validateQueryTypes(query, limits.cardinalColumns,
+                    limits.queryRules.groupBy);
+                case "topn": return this.validateQueryTypes(query, limits.cardinalColumns, limits.queryRules.topN);
+                case "scan": return this.validateQueryTypes(query, limits.cardinalColumns, limits.queryRules.scan);
+                // tslint:disable-next-line:max-line-length
+                case "select": return this.validateQueryTypes(query, limits.cardinalColumns, limits.queryRules.select);
+                // tslint:disable-next-line:max-line-length
+                case "search": return this.validateQueryTypes(query, limits.cardinalColumns, limits.queryRules.search);
+                // tslint:disable-next-line:max-line-length
+                case "timeseries": return this.validateQueryTypes(query, limits.cardinalColumns, limits.queryRules.timeseries);
+                // tslint:disable-next-line:max-line-length
+                case "timeboundary": return this.validateQueryTypes(query, limits.cardinalColumns, limits.queryRules.timeBoundary);
+                // tslint:disable-next-line:max-line-length
+                case "segmentmetadata": return this.validateQueryTypes(query, limits.cardinalColumns, limits.queryRules.segmentMetadata);
+
+                default: return {
+                    error: undefined,
+                    errorMessage: `Unsupported query type"${query.queryType}"`,
+                    isValid: false,
+                };
+            }
```

Comment:
This can be further simplified by the method getRules(queryType) and this method would just be the logic to validate.